doc/languages-frameworks/vscode.section.md

```diff
@@ -0,0 +1,121 @@
+# Visual Studio Code {#visual-studio-code}
+
+_For the free distribution Visual Studio Code (VSC) without MS branding/telemetry see [vscodium](https://nixos.wiki/wiki/VSCodium)_
+
+To install VSC just add it to `systemPackages`:
+
+```nix
+{ environment.systemPackages = with pkgs; [ vscode ]; }
+```
+
+You'll likely want to install some extensions though. For this replace `vscode` with `vscode-with-extensions` and override the `vscodeExtensions` attribute:
+
+```nix
+let
+  extensions = (with pkgs.vscode-extensions; [
+    bbenoist.Nix
+  ]) ++ pkgs.vscode-utils.extensionsFromVscodeMarketplace [{
+    name = "remote-ssh-edit";
+    publisher = "ms-vscode-remote";
+    version = "0.47.2";
+    sha256 = "1hp6gjh4xp2m1xlm1jsdzxw9d8frkiidhph6nvl24d0h8z34w49g";
+  }];
+  vscode-with-extensions = pkgs.vscode-with-extensions.override {
+    vscodeExtensions = extensions;
+  };
+in
+{
+  environment.systemPackages = [
+    vscode-with-extensions
```

Comment:
I would prefer not to use `vscode-with-extensions` twice
```suggestion
  vscode-with-my-extensions = pkgs.vscode-with-extensions.override {
    vscodeExtensions = extensions;
  };
in
{
  environment.systemPackages = [
    vscode-with-my-extensions
```

Reply:
I'll take care of this later today and also add some info about your recent PRs about buildFSHUserEnv for VSCode, thanks fo the review :confetti_ball: